### ISSUE TITLE
Inaccurate Contrastive Loss in the example?

### ISSUE THREAD

**Issue by gkmeng** [keras-team/keras#16935]
Have been following the tutorial below to construct my siamese network. However, I am not sure if the contrastive loss written in this tutorial is correct.

https://keras.io/examples/vision/siamese_contrastive/

In the original paper, similar pair = 0, dissimilar pair = 1, and the eqn of the contrastive loss:

loss = [ (1-y) * D ] + (y * max(margin, D))

However, in the given tutorial, similar pair = 1, dissimilar pair = 0, and the eqn above are still used for calculating the loss. 
Appreciate if anyone can help in clarifying this issue, thank you.

**Comment by tilakrayal** [comment]
@gkmeng,
Could you please feel free to submit a PR for the requested change or share the link where the requested change is to be made and the code to reproduce the issue. Thank you!

**Comment by gkmeng** [comment]
> @gkmeng, Could you please feel free to submit a PR for the requested change or share the link where the requested change is to be made and the code to reproduce the issue. Thank you!

Hi, similar request has been made in the past:

https://github.com/keras-team/keras/pull/13592

**Comment by hertschuh** [comment]
@gkmeng ,

The loss function is fixed and matches the paper:
https://github.com/keras-team/keras-io/blob/master/examples/vision/siamese_contrastive.py#L335

So in that sense, the "similar PR" you mentioned was integrated (but the examples also moved since).

The only discrepancy is how labels are defined. You are correct, that in the original paper, similar pair = 0, dissimilar pair = 1, and in the example, it is reversed. It still works because the model simply learns to output the opposite.

**Comment by hertschuh** [comment]
The fix is now up on keras.io. Closing.
